internal/pkg/gateway/commit/blockparser.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"github.com/golang/protobuf/proto"
+	commonProto "github.com/hyperledger/fabric-protos-go/common"
+	peerProto "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/pkg/errors"
+)
+
+type blockParser struct {
+	Block *commonProto.Block
+}
+
+func (parser *blockParser) TransactionValidationCodes() (map[string]peerProto.TxValidationCode, error) {
+	result := make(map[string]peerProto.TxValidationCode)
+
+	envelopes := parser.Block.Data.Data
+	for i, envelopeBytes := range envelopes {
+		channelHeader, err := unmarshallChannelHeader(envelopeBytes)
```

Comment:
I agree it is better to ignore malformed committed transactions rather than stop with an error. I'll change that. Checking for nil is also a good call so I'll add some unit tests to drive that implementation.

Is it expected that committed blocks can contain malformed transaction protobufs?